python/activator/middleware_interface.py

```diff
@@ -703,26 +710,36 @@ def _export_calibs(self, detector_id, filter):
         """
         # TAI observation start time should be used for calib validity range.
         calib_date = astropy.time.Time(self.visit.private_sndStamp, format="unix_tai")
-        # Querying by specific types is much faster than querying by ...
+        collections_info = self.central_butler.collections.query_info(
+            self.instrument.makeCalibrationCollectionName(), include_summary=True)
         # Some calibs have an exposure ID (of the source dataset?), but these can't be used in AP.
-        types = {t for t in self.central_butler.registry.queryDatasetTypes()
+        types = {t.name for t in self.central_butler.registry.queryDatasetTypes()
                  if t.isCalibration() and "exposure" not in t.dimensions}
+        types = self.central_butler.collections._filter_dataset_types(types, collections_info)
```

Comment:
In a way, we don't _have to_ call it. Not filtering down the dataset types (as in before this commit) just means looping through more dataset types without data to export. 

From the MW team, it sounds like an upcoming new API in the new query system may be useful to this. I'll add a TODO comment with the middleware ticket number.

Reply:
Comments added with two open jira tickets in the comments.

Reply:
FWIW, this loop is (was?) fairly fast, taking less than a second.

Reply:
The filtering was partly because the way I looped through each type and logged with each type, and partly because it seems to me that filtering down can only help and not hurt.  Now that the logs are switched back to the old style (i.e. not logging 0 for non-existing types), having so many types isn't a problem anymore.  But still, it seems a good idea to filter down. This is now in a separate commit https://github.com/lsst-dm/prompt_processing/pull/204/commits/c8c381587903c85e31589eebff36a61fabc7f012